Raven.Client.Lightweight/Listeners/IDocumentSaveChangesListener.cs

```diff
@@ -0,0 +1,20 @@
+//-----------------------------------------------------------------------
+// <copyright file="IDocumentStoreListener.cs" company="Hibernating Rhinos LTD">
+//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+using Raven.Json.Linq;
+
+namespace Raven.Client.Listeners
+{
+    /// <summary>
+    /// Hook for users to provide additional logic on store operations
+    /// </summary>
+    /// 
+    public interface IDocumentSaveChangesListener
+    {
+        void BeforeSaveChanges(IDocumentSession session);
+
+        void AfterSaveChanges(IDocumentSession session);
+    }
```

Comment:
I'm using an extension method on IDocumentSession.Publish(evnt) that stores those on session.Advanced.ExternalState["Events"] as List<Event>, and retrieve them on the listener via
 var events = session.Advanced.ExternalState["Events"] as List<Event>;
https://gist.github.com/valeriob/3ae27778986da42c7a02c144d88ae52e

Basically what I'm doing is customizing IDocumentSession as my UnitOfWork, this can be done both with wrapping the IDocumentSession or with some extension method and IDocumentSession.Advanced.ExternalState.

I'll go forward with IDocumentSession wrapper and i'll bring this back if i hit any walls.

Thanks for now.